### ISSUE TITLE
incommand with `c` flag to `:s`

### ISSUE THREAD

**Issue by AckslD** [neovim/neovim#20259]
### Feature already in Vim?

I don't think so

### Feature description

When writing a substitution we can use `inccommand` to preview the changes in the buffer and/or in a split, which is really great! The if you add the `c` flag to the substitution command to confirm each substitution you no longer have a preview for what the substitution will be when the confirm dialog is active.

I think it would be really useful if one could preview the changes before confirming a substitution, maybe either as virtual text or in the split used by `inccommand`.

Would this make sense? If so, would it fit in core or better in a plugin? I'd be happy to try to work on this if it sounds like a good idea. Although I don't know in detail how to do it yet.

**Comment by bobrippling** [comment]
This should be covered in most cases when we [set `msg_silent`](https://github.com/neovim/neovim/blob/55d346fc2683a18256320eaae75eb4f3137dcced/src/nvim/ex_getln.c#L2454-L2454) during `:s`, as we [won't prompt when that's set](https://github.com/neovim/neovim/blob/55d346fc2683a18256320eaae75eb4f3137dcced/src/nvim/ex_docmd.c#L2127-L2135).

In the confirmation specific case (`:s/.../.../c`), we also [handle `'inccommand'`](https://github.com/neovim/neovim/blob/55d346fc2683a18256320eaae75eb4f3137dcced/src/nvim/ex_cmds.c#L3643-L3643) - this was [originally added in 2016](https://github.com/neovim/neovim/commit/13841a56b4b4916a2f18e66f97cb6f5e55d40e53#diff-79c00103dd452f1d0d57ea0e308f821d6a85d139f2327098e161ff2869bdc52bR3455) (you'll have to expand `ex_cmds.c`) - is this still an issue in your version?

**Comment by AckslD** [comment]
@bobrippling Yes, unless I'm doing something wrong I don't see the preview anymore in the buffer when the prompt to confirm opens:


https://user-images.githubusercontent.com/23341710/233800186-5705df2b-ddc4-4857-bfe3-6ceb03d2bb41.mp4

**Comment by bobrippling** [comment]
I get you now, I thought you mean while typing the `:s`, not after hitting enter on it. Yeah that would likely be a new feature addition, reusing some of the preview code, probably not do-able in a plugin
